swift/extractor/visitors/VisitorBase.h

```diff
@@ -29,7 +29,7 @@ class VisitorBase {
  public:                                                                                          \
   void visit##CLASS##KIND(swift::CLASS##KIND* e) {                                                \
     using TranslateResult = std::invoke_result_t<decltype(&CrtpSubclass::translate##CLASS##KIND), \
-                                                 CrtpSubclass, swift::CLASS##KIND>;               \
+                                                 CrtpSubclass, swift::CLASS##KIND&>;              \
```

Comment:
Is this necessary for the unresolved pattern change, or is this just a bug fix?

Reply:
If I recall correctly, yes, this was required so it could detect that implementation on a non-const reference, but I can double-check

Reply:
yep, code does not compile without this